clientconn.go

```diff
@@ -206,6 +208,15 @@ func WithBalancerBuilder(b balancer.Builder) DialOption {
 	}
 }
 
+// WithBalancerName sets the balancer that the ClientConn will be initialized
+// with. The balancer can not be overridden by balancer option specified by
```

Comment:
Nit: "cannot" or "will not".

Also, maybe something like: "Balancers are registered by name using the balancer package" before this sentence?